### PR TITLE
Implement @Damage inline rolls, which support modifiers and damage dice 

### PR DESCRIPTION
**Pull request by CarlosFdez**

### CODE REVIEW

File: types/foundry/client/core/text-editor.d.ts [hunk 1]
```diff
@@ -140,7 +140,7 @@ declare global {
          * Handle left-mouse clicks on an inline roll, dispatching the formula or displaying the tooltip
          * @param event The initiating click event
          */
-        static _onClickInlineRoll(event: MouseEvent): Promise<ChatMessage>;
```
Comment (stwlam) [inline types/foundry/client/core/text-editor.d.ts:143]: scanning the source, it looks like this should be `Promise<ChatMessage | void>`

Comment (CarlosFdez) [reply types/foundry/client/core/text-editor.d.ts:143]: In terms of what it does maybe, but given this is just a listener the return type doesnt actually matter. Foundry only returns chat message so thst its async i imagine.

Comment (stwlam) [reply types/foundry/client/core/text-editor.d.ts:143]: It's async merely by having an `async` modifier (which isn't seen in the type)

File: src/module/system/damage/helpers.ts [hunk 2]
```diff
@@ -34,6 +44,99 @@ const DamageCategorization = {
     },
 } as const;
 
+const faces = [4, 6, 8, 10, 12];
```
Comment (stwlam) [inline src/module/system/damage/helpers.ts:47]: All caps? `faces` is shadowed immediately below, which is confusing.

Comment (CarlosFdez) [reply src/module/system/damage/helpers.ts:47]: Good catch, I hadn't noticed that.

**Pull request merged**